spec/support_spec/mocked_permission_helper_spec.rb

```diff
@@ -0,0 +1,170 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+
+RSpec.describe MockedPermissionHelper do
```

Comment:
I'd add some association defaults since we're creating work packages for every example and there will be some factory cascading otherwise
 
```suggestion
RSpec.describe MockedPermissionHelper do
  shared_association_default(:author, factory_name: :user) { create(:user) }
  shared_association_default(:priority) { create(:priority) }
  shared_association_default(:status) { create(:status) }
```

(These should work)

Reply:
I switched to `build` strategy, as the mocking is explicitly for the cases where we only build :)